pkg/distributors/contracts/RewardsScheduler.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/SafeERC20.sol";
+import "./interfaces/IMultiRewards.sol";
+
+// solhint-disable not-rely-on-time
+
+/**
+ * Scheduler for MultiRewards contract
+ */
+contract RewardsScheduler {
+    using SafeERC20 for IERC20;
+
+    IMultiRewards private immutable _multirewards;
+
+    constructor() {
+        _multirewards = IMultiRewards(msg.sender);
+    }
+
+    enum RewardStatus { UNINITIALIZED, PENDING, STARTED }
+
+    struct ScheduledReward {
+        IERC20 pool;
+        IERC20 rewardsToken;
+        uint256 startTime;
+        address rewarder;
+        uint256 amount;
+        RewardStatus status;
+    }
+
+    event RewardScheduled(
+        bytes32 rewardId,
+        address indexed scheduler,
+        IERC20 indexed pool,
+        IERC20 indexed rewardsToken,
+        uint256 startTime,
+        uint256 amount
+    );
+    event RewardStarted(
+        bytes32 rewardId,
+        address indexed scheduler,
```

Comment:
It seems like this 'scheduler' is called 'rewarder' everywhere else.

Reply:
So actually in this case the param is referring to the user that actually started the reward (the person that poked the contract) but I think on second thought it does make more sense to report the `rewarder` here, so I have changed it